engine/transaction/record/record_transaction.go

```diff
@@ -4,44 +4,40 @@ import (
 	"context"
 
 	trx "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
 )
 
-// RecordTransaction will validate, broadcast and save a transaction
-func (s *Service) RecordTransaction(ctx context.Context, tx *trx.Transaction, verifyScripts bool) error {
+// RecordPaymailTransaction will validate, broadcast and save paymail transaction
+func (s *Service) RecordPaymailTransaction(ctx context.Context, tx *trx.Transaction, senderPaymail, receiverPaymail string) error {
 	flow := newTxFlow(ctx, s, tx)
 
-	if verifyScripts {
-		// TODO: Check if in case of not-veryfying-scripts we accidentally allow removing UserUTXOs from the database
-		// NOTE: When we want to record "RawTX" we cannot verify scripts
-		if err := flow.verifyScripts(); err != nil {
-			return err
-		}
-	}
-
-	utxosToSpend, trackedOutputs, err := flow.getFromInputs()
+	trackedOutputs, err := flow.getFromInputs()
 	if err != nil {
 		return err
 	}
 
-	for _, utxo := range utxosToSpend {
-		operation := flow.operationOfUser(utxo.UserID)
+	for _, utxo := range trackedOutputs {
+		operation := flow.operationOfUser(utxo.UserID, "outgoing", receiverPaymail)
+		if len(flow.operations) > 1 {
+			return spverrors.Newf("paymail transaction with multiple senders is not supported")
+		}
 		operation.subtract(utxo.Satoshis)
 	}
 
 	flow.spendInputs(trackedOutputs)
 
-	newOutputs, err := flow.getOutputsForTrackedAddresses()
+	p2pkhOutputs, err := flow.findRelevantP2PKHOutputs()
 	if err != nil {
 		return err
 	}
 
-	for output := range newOutputs {
-		utxo := output.ToUserUTXO()
-		if utxo != nil {
-			operation := flow.operationOfUser(utxo.UserID)
-			operation.add(utxo.Satoshis)
+	for outputData := range p2pkhOutputs {
+		operation := flow.operationOfUser(outputData.userID, "incoming", senderPaymail)
```

Comment:
Just raising some concern - I don't know if this is possible to be fixed.
It is a little confusing that we're getting the operation saying that it is incoming, but if there is already outgoing operation then we will get the outgoing operation.

I know that calculation will be ok. And I don't have idea yet how to make it better. But just putting it here, maybe someone will find some solution.

Reply:
let's move it for later stage. Maybe we will found some solution